diff -up onnxruntime-1.21.1/cmake/CMakeLists.txt.1~ onnxruntime-1.21.1/cmake/CMakeLists.txt
--- onnxruntime-1.21.1/cmake/CMakeLists.txt.1~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/CMakeLists.txt	2025-10-05 19:40:07.539171721 +0200
@@ -88,6 +88,7 @@ option(onnxruntime_USE_RKNPU "Build with
 option(onnxruntime_USE_DNNL "Build with DNNL support" OFF)
 option(onnxruntime_USE_JSEP "Build with JavaScript implemented kernels support" OFF)
 option(onnxruntime_BUILD_UNIT_TESTS "Build ONNXRuntime unit tests" ON)
+option(onnxruntime_INSTALL_UNIT_TESTS "Install ONNXRuntime unit tests" ON)
 option(onnxruntime_BUILD_CSHARP "Build C# library" OFF)
 option(onnxruntime_BUILD_OBJC "Build Objective-C library" OFF)
 option(onnxruntime_USE_PREINSTALLED_EIGEN "Use pre-installed EIGEN. Need to provide eigen_SOURCE_PATH if turn this on." OFF)
diff -up onnxruntime-1.21.1/cmake/onnxruntime_unittests.cmake.1~ onnxruntime-1.21.1/cmake/onnxruntime_unittests.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_unittests.cmake.1~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_unittests.cmake	2025-10-05 19:42:34.736789496 +0200
@@ -1130,11 +1130,13 @@ if (NOT IOS)
     endif()
     set_target_properties(onnx_test_runner PROPERTIES FOLDER "ONNXRuntimeTest")
 
-    install(TARGETS onnx_test_runner
-            ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            BUNDLE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
+    if (onnxruntime_INSTALL_UNIT_TESTS)
+        install(TARGETS onnx_test_runner
+                ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                BUNDLE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
+    endif()
 endif()
 
 if (NOT onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
