diff -up onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake.6~ onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake
--- onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake.6~	2025-10-05 19:48:29.555640357 +0200
+++ onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake	2025-10-05 19:49:08.410660267 +0200
@@ -514,15 +514,6 @@ else()
   include(external/eigen.cmake)
 endif()
 
-if(WIN32)
-  if(onnxruntime_USE_VCPKG)
-    find_package(wil CONFIG REQUIRED)
-    set(WIL_TARGET "WIL::WIL")
-  else()
-    include(wil) # FetchContent
-  endif()
-endif()
-
 # XNNPACK EP
 if (onnxruntime_USE_XNNPACK)
   if (onnxruntime_DISABLE_CONTRIB_OPS)
@@ -547,7 +538,7 @@ if (onnxruntime_USE_XNNPACK)
   endif()
 endif()
 
-set(onnxruntime_EXTERNAL_LIBRARIES ${onnxruntime_EXTERNAL_LIBRARIES_XNNPACK} ${WIL_TARGET} nlohmann_json::nlohmann_json
+set(onnxruntime_EXTERNAL_LIBRARIES ${onnxruntime_EXTERNAL_LIBRARIES_XNNPACK} nlohmann_json::nlohmann_json
                                    onnx onnx_proto ${PROTOBUF_LIB} re2::re2 Boost::mp11 safeint_interface
                                    flatbuffers::flatbuffers ${GSL_TARGET} ${ABSEIL_LIBS} date::date
                                    ${ONNXRUNTIME_CLOG_TARGET_NAME} Eigen3::Eigen)
diff -up onnxruntime-1.21.1/cmake/onnxruntime_common.cmake.6~ onnxruntime-1.21.1/cmake/onnxruntime_common.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_common.cmake.6~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_common.cmake	2025-10-05 19:49:35.938934724 +0200
@@ -127,7 +127,7 @@ if (MSVC)
     endif()
 endif()
 
-onnxruntime_add_include_to_target(onnxruntime_common date::date ${WIL_TARGET} Eigen3::Eigen)
+onnxruntime_add_include_to_target(onnxruntime_common date::date Eigen3::Eigen)
 target_include_directories(onnxruntime_common
     PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${ONNXRUNTIME_ROOT}
     # propagate include directories of dependencies that are part of public interface
diff -up onnxruntime-1.21.1/cmake/onnxruntime_graph.cmake.6~ onnxruntime-1.21.1/cmake/onnxruntime_graph.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_graph.cmake.6~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_graph.cmake	2025-10-05 19:48:29.566734099 +0200
@@ -96,7 +96,7 @@ endif()
 
 onnxruntime_add_static_library(onnxruntime_graph ${onnxruntime_graph_src} ${orttraining_graph_src})
 add_dependencies(onnxruntime_graph onnx_proto flatbuffers::flatbuffers)
-onnxruntime_add_include_to_target(onnxruntime_graph onnxruntime_common ${WIL_TARGET} onnx onnx_proto ${PROTOBUF_LIB} flatbuffers::flatbuffers safeint_interface Boost::mp11)
+onnxruntime_add_include_to_target(onnxruntime_graph onnxruntime_common onnx onnx_proto ${PROTOBUF_LIB} flatbuffers::flatbuffers safeint_interface Boost::mp11)
 
 if (MSVC)
   set(ONNX_PROTOBUF_NATVIS_FILE "onnx_protobuf.natvis")
diff -up onnxruntime-1.21.1/cmake/onnxruntime_providers_dml.cmake.6~ onnxruntime-1.21.1/cmake/onnxruntime_providers_dml.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_providers_dml.cmake.6~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_providers_dml.cmake	2025-10-05 19:49:54.128155267 +0200
@@ -9,7 +9,7 @@
   source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_dml_cc_srcs})
   onnxruntime_add_static_library(onnxruntime_providers_dml ${onnxruntime_providers_dml_cc_srcs})
   onnxruntime_add_include_to_target(onnxruntime_providers_dml
-    onnxruntime_common onnxruntime_framework onnx onnx_proto ${PROTOBUF_LIB} flatbuffers::flatbuffers Boost::mp11 safeint_interface ${WIL_TARGET}
+    onnxruntime_common onnxruntime_framework onnx onnx_proto ${PROTOBUF_LIB} flatbuffers::flatbuffers Boost::mp11 safeint_interface
   )
   if(TARGET Microsoft::DirectX-Headers)
     onnxruntime_add_include_to_target(onnxruntime_providers_dml Microsoft::DirectX-Headers)
diff -up onnxruntime-1.21.1/cmake/winml_unittests.cmake.6~ onnxruntime-1.21.1/cmake/winml_unittests.cmake
--- onnxruntime-1.21.1/cmake/winml_unittests.cmake.6~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/winml_unittests.cmake	2025-10-05 19:48:29.566832565 +0200
@@ -186,7 +186,7 @@ add_dependencies(winml_test_common
   winml_dll
 )
 
-onnxruntime_add_include_to_target(winml_test_common onnx_proto GTest::gtest ${PROTOBUF_LIB} ${WIL_TARGET} safeint_interface ${GSL_TARGET})
+onnxruntime_add_include_to_target(winml_test_common onnx_proto GTest::gtest ${PROTOBUF_LIB} safeint_interface ${GSL_TARGET})
 onnxruntime_add_static_library(winml_google_test_lib ${WINML_TEST_SRC_DIR}/common/googletest/main.cpp)
 onnxruntime_add_include_to_target(winml_google_test_lib GTest::gtest)
 set_winml_target_properties(winml_google_test_lib)
