diff -up onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake.7~ onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake
--- onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake.7~	2025-10-05 19:50:09.917354927 +0200
+++ onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake	2025-10-05 19:51:00.048588050 +0200
@@ -398,21 +398,8 @@ set(GSL_TARGET "Microsoft.GSL::GSL")
 set(GSL_INCLUDE_DIR "$<TARGET_PROPERTY:${GSL_TARGET},INTERFACE_INCLUDE_DIRECTORIES>")
 onnxruntime_fetchcontent_makeavailable(GSL)
 
-find_path(safeint_SOURCE_DIR NAMES "SafeInt.hpp")
-if(NOT safeint_SOURCE_DIR)
-  unset(safeint_SOURCE_DIR)
-  onnxruntime_fetchcontent_declare(
-      safeint
-      URL ${DEP_URL_safeint}
-      URL_HASH SHA1=${DEP_SHA1_safeint}
-      EXCLUDE_FROM_ALL
-  )
-
-  # use fetch content rather than makeavailable because safeint only includes unconditional test targets
-  FetchContent_Populate(safeint)
-endif()
 add_library(safeint_interface INTERFACE)
-target_include_directories(safeint_interface INTERFACE ${safeint_SOURCE_DIR})
+target_include_directories(safeint_interface INTERFACE /usr/include/SafeInt)
 
 
 # Flatbuffers
diff -up onnxruntime-1.21.1/onnxruntime/core/common/safeint.h.7~ onnxruntime-1.21.1/onnxruntime/core/common/safeint.h
--- onnxruntime-1.21.1/onnxruntime/core/common/safeint.h.7~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/onnxruntime/core/common/safeint.h	2025-10-05 19:50:09.923752504 +0200
@@ -24,6 +24,7 @@ class SafeIntExceptionHandler<onnxruntim
 
 #define SAFEINT_EXCEPTION_HANDLER_CPP 1
 #define SafeIntDefaultExceptionHandler SafeIntExceptionHandler<onnxruntime::OnnxRuntimeException>
+#define SAFEINT_CPP_THROW noexcept(false)
 
 #if defined(__GNUC__)
 #include "onnxruntime_config.h"
