diff -up onnxruntime-1.21.1/cmake/onnxruntime_providers_cpu.cmake.8~ onnxruntime-1.21.1/cmake/onnxruntime_providers_cpu.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_providers_cpu.cmake.8~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_providers_cpu.cmake	2025-10-05 19:51:54.021387955 +0200
@@ -229,6 +229,7 @@ if (NOT onnxruntime_MINIMAL_BUILD AND NO
   onnxruntime_add_shared_library(onnxruntime_providers_shared ${onnxruntime_providers_shared_cc_srcs} "${ONNXRUNTIME_ROOT}/core/dll/onnxruntime.rc")
   set_target_properties(onnxruntime_providers_shared PROPERTIES FOLDER "ONNXRuntime")
   set_target_properties(onnxruntime_providers_shared PROPERTIES LINKER_LANGUAGE CXX)
+  set_target_properties(onnxruntime_providers_shared PROPERTIES VERSION ${ORT_VERSION})
   #ON AIX, to call dlopen on onnxruntime_providers_shared, we need to generate this library as shared object .so file.
   #Latest cmake behavior is changed and  cmake will remove shared object .so file after generating the shared archive.
   #To prevent that, making AIX_SHARED_LIBRARY_ARCHIVE as OFF for onnxruntime_providers_shared.
diff -up onnxruntime-1.21.1/cmake/onnxruntime_providers_cuda.cmake.8~ onnxruntime-1.21.1/cmake/onnxruntime_providers_cuda.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_providers_cuda.cmake.8~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_providers_cuda.cmake	2025-10-05 19:51:16.685820703 +0200
@@ -239,6 +239,7 @@
     # ${CMAKE_CURRENT_BINARY_DIR} is so that #include "onnxruntime_config.h" inside tensor_shape.h is found
     set_target_properties(${target} PROPERTIES LINKER_LANGUAGE CUDA)
     set_target_properties(${target} PROPERTIES FOLDER "ONNXRuntime")
+    set_target_properties(${target} PROPERTIES VERSION ${ORT_VERSION})
 
     if (onnxruntime_ENABLE_CUDA_PROFILING) # configure cupti for cuda profiling
       target_link_libraries(${target} PRIVATE CUDA::cupti)
diff -up onnxruntime-1.21.1/cmake/onnxruntime_providers_rocm.cmake.8~ onnxruntime-1.21.1/cmake/onnxruntime_providers_rocm.cmake
--- onnxruntime-1.21.1/cmake/onnxruntime_providers_rocm.cmake.8~	2025-04-18 21:12:09.000000000 +0200
+++ onnxruntime-1.21.1/cmake/onnxruntime_providers_rocm.cmake	2025-10-05 19:51:16.685891777 +0200
@@ -154,6 +154,7 @@
 
   set_target_properties(onnxruntime_providers_rocm PROPERTIES LINKER_LANGUAGE CXX)
   set_target_properties(onnxruntime_providers_rocm PROPERTIES FOLDER "ONNXRuntime")
+  set_target_properties(onnxruntime_providers_rocm PROPERTIES VERSION ${ORT_VERSION})
   target_compile_definitions(onnxruntime_providers_rocm PRIVATE HIPBLAS_V2)
 
   if (onnxruntime_ENABLE_TRAINING)
