diff -up onnxruntime-1.23.0/cmake/external/abseil-cpp.cmake.4~ onnxruntime-1.23.0/cmake/external/abseil-cpp.cmake
--- onnxruntime-1.23.0/cmake/external/abseil-cpp.cmake.4~	2025-10-06 03:10:25.906820100 +0200
+++ onnxruntime-1.23.0/cmake/external/abseil-cpp.cmake	2025-10-06 03:10:25.984073666 +0200
@@ -122,7 +122,6 @@ absl::absl_check
 absl::hash_function_defaults
 absl::function_ref
 absl::city
-absl::low_level_hash
 absl::fixed_array
 absl::variant
 absl::meta
diff -up onnxruntime-1.23.0/cmake/onnxruntime_webassembly.cmake.4~ onnxruntime-1.23.0/cmake/onnxruntime_webassembly.cmake
