diff -up onnxruntime-1.22.2/cmake/external/onnxruntime_external_deps.cmake.4~ onnxruntime-1.22.2/cmake/external/onnxruntime_external_deps.cmake
--- onnxruntime-1.22.2/cmake/external/onnxruntime_external_deps.cmake.4~	2025-10-06 02:15:02.247982092 +0200
+++ onnxruntime-1.22.2/cmake/external/onnxruntime_external_deps.cmake	2025-10-06 02:15:31.432048378 +0200
@@ -534,11 +534,7 @@ if(TARGET ONNX::onnx_proto AND NOT TARGE
   message(STATUS "Aliasing ONNX::onnx_proto to onnx_proto")
   add_library(onnx_proto ALIAS ONNX::onnx_proto)
 endif()
-if(onnxruntime_USE_VCPKG)
-  find_package(Eigen3 CONFIG REQUIRED)
-else()
-  include(external/eigen.cmake)
-endif()
+find_package(Eigen3 CONFIG REQUIRED)
 
 if(WIN32)
   if(onnxruntime_USE_VCPKG)
