diff -up onnxruntime-1.23.0/onnxruntime/core/common/spin_pause.cc.omv~ onnxruntime-1.23.0/onnxruntime/core/common/spin_pause.cc
--- onnxruntime-1.23.0/onnxruntime/core/common/spin_pause.cc.omv~	2025-10-06 03:39:36.295748982 +0200
+++ onnxruntime-1.23.0/onnxruntime/core/common/spin_pause.cc	2025-10-06 03:41:55.779619158 +0200
@@ -35,7 +35,12 @@ void SpinPause() {
 #if defined(_WIN32)
     _tpause(0x0, __rdtsc() + tpause_spin_delay_cycles);
 #elif defined(__linux__)
+#ifdef __clang__
+    const uint64_t cycles = __rdtsc() + tpause_spin_delay_cycles;
+    __builtin_ia32_tpause(0x0, static_cast<uint32_t>(cycles>>32), static_cast<uint32_t>(cycles));
+#else
     __builtin_ia32_tpause(0x0, __rdtsc() + tpause_spin_delay_cycles);
+#endif
 #else
     _mm_pause();
 #endif
