diff -up onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake.omv~ onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake
--- onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake.omv~	2025-10-05 20:19:30.722678664 +0200
+++ onnxruntime-1.21.1/cmake/external/onnxruntime_external_deps.cmake	2025-10-05 20:19:43.352786140 +0200
@@ -429,7 +429,7 @@ namespace std { using ::getenv; }
     endif()
   endif()
 else()
-  add_library(flatbuffers::flatbuffers ALIAS flatbuffers::flatbuffers_shared)
+	#add_library(flatbuffers::flatbuffers ALIAS flatbuffers::flatbuffers_shared)
 endif()
 
 # ONNX
